glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -36,53 +27,32 @@ import java.util.concurrent.TimeUnit as AndroidTimeUnit
  * - ping is overdue (due time already passed) for the current calendar day;
  * - ping is soon to be sent in the current calendar day;
  * - ping was already sent, and must be scheduled for the next calendar day.
- *
- * The scheduler also makes use of the [LifecycleObserver] in order to correctly schedule
- * the [MetricsPingWorker].
  */
 @Suppress("TooManyFunctions")
 internal class MetricsPingScheduler(
     private val applicationContext: Context,
     migratedLastSentDate: String? = null
-) : LifecycleEventObserver {
+) {
     internal val sharedPreferences: SharedPreferences by lazy {
         applicationContext.getSharedPreferences(this.javaClass.canonicalName, Context.MODE_PRIVATE)
     }
 
+    internal var timer: Timer? = null
+
     companion object {
         private const val LOG_TAG = "glean/MetricsPingSched"
         const val LAST_METRICS_PING_SENT_DATETIME = "last_metrics_ping_iso_datetime"
         const val DUE_HOUR_OF_THE_DAY = 4
-        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-        internal var isInForeground = false
-        internal var firstForeground = true
-
-        /**
-         * Function to cancel any pending metrics ping workers
-         */
-        internal fun cancel(context: Context) {
-            WorkManager.getInstance(context).cancelUniqueWork(MetricsPingWorker.TAG)
```

Comment:
I wonder if we should keep some code in the init to cancel any outstanding unique work: I'm not sure they get cleared automatically. What do you think? Or maybe we could land some metric to check if we find any existing work after the upgrade.

Reply:
From reading the API docs, you can cancel a scheduled timer, but you can't cancel work already in progress (i.e. once the timer has triggered, but before its task function has completed).  I've been careful in this PR to always cancel the timer before scheduling new work, so there will always only be one thing "pending" at any given time.  Is there something else you're suggesting?

Reply:
Sorry, I was unclear. I'm not concerned about cancelling the _timers_ we're now introducing. I'm concerned about cancelling any outstanding `WorkManager` job that was enqueued in previous versions of the Glean SDK.

I would not want `WorkManager` jobs to keep living forever. This shouldn't be the case, as they are not recurring jobs, but the `WorkManager` has the habit of surprising us every now and then :)

Reply:
Ah.  That's crazy town, but we know how crazy Android can be :)

I really hope this isn't possible (since it would have to run code from a previous revision of the app, which is hopefully completely gone), but I agree, to be extra sure, we should probably keep something like this in.